integration/examples/custom-buildx/Dockerfile

```
@@ -0,0 +1,12 @@
+FROM golang:1.16-alpine as builder
+COPY main.go .
+# `skaffold debug` sets SKAFFOLD_GO_GCFLAGS to disable compiler optimizations
+ARG SKAFFOLD_GO_GCFLAGS
+RUN go build -gcflags="${SKAFFOLD_GO_GCFLAGS}" -o /app main.go
```

should we use `TARGETARCH` and `TARGETOS`  args here and pass them to `go build`
```suggestion
FROM --platform=$BUILDPLATFORM golang:alpine AS build                                                                       
                                                                       
ARG TARGETOS                                                                                                                
ARG TARGETARCH                                                                                                                                                                                          
RUN GOOS=$TARGETOS GOARCH=$TARGETARCH  go build -gcflags="${SKAFFOLD_GO_GCFLAGS}" -o /app main.go 
```